src/ErrorProne.NET/ErrorProne.NET.Test/ExceptionHandlingRules/ArgumentExceptionUsesStringAnalyzerTests.cs

```diff
@@ -0,0 +1,154 @@
+using ErrorProne.NET.Rules.ExceptionHandling;
+using NUnit.Framework;
+using RoslynNunitTestRunner;
+
+namespace ErrorProne.NET.Test.ExceptionHandling
+{
+    [TestFixture]
+    public class ArgumentExceptionUsesStringAnalyzerTests : CSharpAnalyzerTestFixture<ArgumentExceptionUsesStringAnalyzer> {
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasNoParametersInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw [|new ArgumentException()|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasNoParametersInMethod() {
+            var code = @"
+using System;
+class Test
+{
+    public Foo(string arg) 
+    {
+        throw [|new ArgumentException()|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasOnlyMessageParameterInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw [|new ArgumentException(""It's an argument exception"")|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasOnlyMessageParameterInMethod() {
+            var code = @"
+using System;
+class Test
+{
+    public Foo(string arg) 
+    {
+        throw [|new ArgumentException(""It's an argument exception"")|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasWrongParameterNameInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", [|""arg1""|]);
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionMethodHasNoSuchParamName);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasWrongParameterNameInMethod() {
+            var code = @"
+using System;
+class Test
+{
+    public Foo(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", [|""arg1""|]);
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionMethodHasNoSuchParamName);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasStringParameterNameInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", [|""arg""|]);
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameShouldNotBeString);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasStringParameterNameInMethod() {
```

Comment:
I can see at least 4 different cases here:
- `throw new ArgumentException("name", "message")` - your case
- `throw new ArgumentException("name", "name")` - the name of the argument is used as description (probably valid). Also, it can be `throw new ArgumentException("name", nameof(name))
- `throw new ArgumentException("name1", "name2")` - the name of one argument is used as description, the other one is used as parameter (both are valid names)
- `throw new ArgumentException("name", variable)` - variable can provide required name

There may be more, but even those add 4 additional rules. Looks more like additional task to me.